first_ruby/app/models/application_record.rb

```diff
@@ -0,0 +1,3 @@
+class ApplicationRecord < ActiveRecord::Base
+  self.abstract_class = true
```

Comment:
What is `self.abstract_class` means?

Reply:
This means that the class ApplicationRecord is declared as an abstract class

Reply:
....
I know that. i am asking wtf is `abstract_class`

Reply:
Abstract classes cannot be instantiated and it can be inherited by child subclass. It can have some methods inside the abstract class, and it is for the child subclass to declare the implementation.

Reply:
ok.
Given that 1 model class represent a table in the database, how does that apply here?

Reply:
The model class will inherit the `Application Record` for example:
```
class Movie < ApplicationRecord 

end
```
This will allow `ORM`(Object Relational Mapping) which is to connect the model and the database table. 
It uses a naming conventions between the mapping of model and database table, it will pluralize the class model name to find the database table. In this case here, it will find the database table called `movies`.

Reply:
huh?
Reread my question please

Reply:
When `self.abstract_class = true` is defined in a model class, it will not have a table.